pypeit/flatfield.py

```diff
@@ -190,144 +193,90 @@ def build_pixflat(self, trim=True, force=False):
-                self.tslits_dict['slit_righ_tweak'][:,slit] = slit_righ_out
-                final_tilts[thismask_out] = tilts_out[thismask_out]
-
-        # If we tweaked the slits update the tilts_dict
-        if self.flatpar['tweak_slits']:
-            self.tilts_dict['tilts'] = final_tilts
+        # Fit it
+        # NOTE: Tilts do not change and self.slits is updated
+        # internally.
+        self.fit(debug=debug)
 
         if show:
             # Global skysub is the first step in a new extraction so clear the channels here
             self.show(slits=True, wcs_match = True)
 
-        # If illumination flat fielding is turned off, set the illumflat to be None.
-        if not self.flatpar['illumflatten']:
-            msgs.warn('No illumination flat will be applied to your data (illumflatten=False).')
-            self.msillumflat = None
-
         # Return
         return self.mspixelflat, self.msillumflat
 
-    def show(self, slits=True, wcs_match=True):
+    def show(self, show_slits=True, wcs_match=True):
         """
-        Show all of the flat field products
+        Show all of the flat field products in ginga.
 
         Args:
-            slits (bool, optional):
-            wcs_match (bool, optional):
-
+            show_slits (:obj:`bool`, optional):
+                Overlay the slit edges on the flat-field images
+            wcs_match (:obj:`bool`, optional):
+                Match the WCS of the flat-field images
         """
+        ginga.connect_to_ginga(raise_err=True, allow_new=True)
+
+        # TODO: Add an option that shows the relevant stuff in a
+        # matplotlib window.
         viewer, ch = ginga.show_image(self.mspixelflat, chname='pixeflat', cuts=(0.9, 1.1),
                                       wcs_match=wcs_match, clear=True)
         viewer, ch = ginga.show_image(self.msillumflat, chname='illumflat', cuts=(0.9, 1.1),
                                       wcs_match=wcs_match)
-        viewer, ch = ginga.show_image(self.rawflatimg, chname='flat', wcs_match=wcs_match)
+        viewer, ch = ginga.show_image(self.rawflatimg.image, chname='flat', wcs_match=wcs_match)
         viewer, ch = ginga.show_image(self.flat_model, chname='flat_model', wcs_match=wcs_match)
 
-        if slits:
-            if self.tslits_dict is not None:
-                slit_ids = [edgetrace.get_slitid(self.rawflatimg.shape,
-                                                 self.tslits_dict['slit_left'],
-                                                 self.tslits_dict['slit_righ'], ii)[0]
-                                for ii in range(self.tslits_dict['slit_left'].shape[1])]
-                ginga.show_slits(viewer, ch, self.tslits_dict['slit_left'],
-                                 self.tslits_dict['slit_righ'], slit_ids)
+        if not show_slits:
+            return
+
+        # Get the slits
+        if self.slits is None:
+            # Try to load the slits
+            try:
```

Comment:
I think that this never faults if the master file does not exist since it returns None. So I don't believe this try except works or at least it never prints out the warning.

Reply:
Fixed in next push.